app/services/update_stacks_files_service.rb

```diff
@@ -38,21 +38,18 @@ def check_signed_ids
 
   # Copy the files from ActiveStorage to the Stacks directory
   def shelve_files
+    make_shelving_dir(stacks_druid_path)
     file_uploads_map.each do |filename, signed_id|
       blob = blob_for_signed_id(signed_id, filename)
       blob_path = ActiveStorage::Blob.service.path_for(blob.key)
 
       shelving_path = File.join(stacks_druid_path, filename)
-      make_shelving_dir(shelving_path)
       FileUtils.cp(blob_path, shelving_path)
     end
   end
 
-  def make_shelving_dir(shelving_path)
-    shelving_dir = File.dirname(shelving_path)
-    return if File.directory?(shelving_dir)
-
-    FileUtils.mkdir_p(shelving_dir)
+  def make_shelving_dir(shelving_dir)
+    File.directory?(shelving_dir) || FileUtils.mkdir_p(shelving_dir)
```

Comment:
Won't `FileUtils.mkdir_p(shelving_dir)` internally no-op if `File.directory?(shelving_dir)` returns `true`? In other words, can we ditch the check? Surprised this didn't trip Rubocop's `Lint/NonAtomicFileOperation` cop.

Reply:
I think so. I'll try that.